src/wallet/wallet.cpp

```diff
@@ -2583,6 +2583,16 @@ void CWallet::UpdatedTransaction(const uint256 &hashTx)
     }
 }
 
+void CWallet::GetScriptForMining(CScript &script)
+{
+    CReserveKey reservekey(this);
```

Comment:
Why not make CReserveKey (or e.g. CReserveScript) a generic base class defined outside the wallet, and return that from `CWallet::GetScriptForMining`, so that like now the caller can decide to keep or return it

This also interacts better with the case where multiple wallets return a script for mining and some have to be discarded. At least with a CReserveKey-like class, these keys will be recycled.